ttnn/cpp/ttnn/operations/eltwise/quantization/quantization.cpp

```diff
@@ -89,8 +242,65 @@ Tensor DequantOp::invoke(
     const std::optional<MemoryConfig>& memory_config,
     std::optional<Tensor> optional_output_tensor) {
     const ttnn::DataType a_dtype = input_tensor.get_dtype();
-    const bool typecast_a = needs_typecast_to_bfloat16(a_dtype);
-    Tensor input_a = typecast_a ? typecast_to(DataType::BFLOAT16, input_tensor) : input_tensor;
+    ttnn::DataType c_dtype = ttnn::DataType::BFLOAT16;
+    if (output_dtype.has_value() && optional_output_tensor.has_value()) {
+        TT_FATAL(
+            output_dtype.value() == optional_output_tensor->dtype(),
+            "Mismatching output_dtype and output tensor dtype");
+        c_dtype = output_dtype.value();
+    } else if (output_dtype.has_value()) {
+        c_dtype = output_dtype.value();
+    } else if (optional_output_tensor.has_value()) {
+        c_dtype = optional_output_tensor->dtype();
+    }
+    TT_FATAL(a_dtype == ttnn::DataType::INT32, "Dequantize only supports int32 inputs for now");
+
+    tt::stl::Span<const ttnn::operations::unary::UnaryWithParam> lhs_activations{};
+    tt::stl::Span<const ttnn::operations::unary::UnaryWithParam> rhs_activations{};
+
+    // LLK dequant kernel does addition, so we need to negate zero_point
+    std::array<const ttnn::operations::unary::UnaryWithParam, 1> post_activations{
```

Comment:
It is atypical for `std::array` to be passed a `const`-qualified value type, instead the array itself should be `const`-qualified. But starting from C++17 you actually don't need these template parameters at all though and can use CTAD (constructor template argument deduction) instead like this:
```suggestion
    const std::array post_activations{
```

Reply:
Done